### DISCUSSION TITLE
Auto-center & -zoom to GeoJSON features possible?

### DISCUSSION THREAD

**Discussion by Moonbase59** [javalent/obsidian-leaflet#110] [Ideas]
If you have many notes showing interesting GeoJSON-coded areas, like @oyvindsolstad’s Interstates, or in my case, countries with states and districts, it’s quite some work to figure the correct map center and zoom.

Usability stuff again, hee hee. I was wondering if it could be possible to

- collect _all_ GeoJSON FeatureCollections (Features) given in a map’s code block,
- determine the common _center_ and _bounding box_, then
- autocenter and autozoom the map based on that
- _if_ a special variable in the code block was set to truthy, like `zoomFeatures: true`?

This would allow to much more easily create notes with maps showing GeoJSON-coded areas, much like some online services do.

I found an [interesting idea on StackOverflow](https://stackoverflow.com/a/65931657), maybe some of this us usable?

What do you think about this idea?

**Comment by valentine195** [comment]
`zoomFeatures` added in 3.18.3

**Comment by Moonbase59** [comment]
Thank you! This feature makes it even more useful, for instance here, where I <kbd>Ctrl</kbd>+<kbd>Click</kbd>’ed on part of the _Route 66_:

![Knowledgebase - Obsidian v0 12 4_154](https://user-images.githubusercontent.com/3706922/122440888-4aff6a00-cf9d-11eb-9531-dd68f2b4df34.png)

**Reply by zhubib1234** [reply]
I never knew how to use geojson files.Can you share the geojson file of Route 66? thank you!
